python/cudf/dataframe/dataframe.py

```diff
@@ -1068,6 +1068,17 @@ def fix_name(name, suffix):
         # Columns are returned in order left - on - right from libgdf
         # Creating dataframe with ordering as pandas:
 
+        dummy = pd.DataFrame(data=dict([(c, [None]) for c in self.columns]),
+                             columns=self.columns) \
+                  .merge(pd.DataFrame(data=dict([(c, [None])
+                                                 for c in other.columns]),
+                                      columns=other.columns),
+                         on=on, how=how,
+                         suffixes=(lsuffix, rsuffix))
```

Comment:
Is the ordering of columns after a merge in Pandas not some type of simple pattern? Having to run a Pandas merge just to get the order seems unnecessarily expensive (even on empty data).

Reply:
The merge column pattern depends on the input parameters. Reading pandas merge source code does not reveal what would be the generic pattern. This does not mean that it would not exist but it is not immediately obvious to me yet.

Btw, merge with empty data gives different ordering from the merge with one-row data (empty data could be a corner case though).

The cost of pandas merge (here about 2ms) is about 2x the cost of creating pandas Dataframe with one row. So, the cost of creating the `dummy` object is about 4x the cost of creating a one-row pandas Dataframe.

Reply:
From pandas gitter I got the following feedback:
```
Q: Is the order of columns as returned by Dataframe.merge predictable?
What would be the corresponding pattern for arbitrary (allowed) combinations
of on, how, and suffixes arguments?
A: normally I would expect merge to preserve the original order of the columns
(as much as possible of course). So first all columns of the left frame, including
the key column in its original order, and then all columns of the right dataframe,
excluding the key column. And for suffixes it doesn't matter, they are added to
the column name in  its original place.
```
I'll update the PR accordingly.